oceannavigator/frontend/src/components/AreaWindow.jsx

```diff
@@ -12,465 +12,355 @@ import CustomPlotLabels from "./CustomPlotLabels.jsx";
     }
-  }
+  };
 
-  onLocalUpdate(key, value) {
-    if (this._mounted) {
-      if (key === "dataset_0") {
-        this.setState((prevState) => ({
-          dataset_0: {
-            ...prevState.dataset_0,
-            ...value,
-          },
-        }));
-        return;
-      }
-
-      if (key === "dataset_1") {
-        this.setState((prevState) => ({
-          dataset_1: {
-            ...prevState.dataset_1,
-            ...value,
-          },
-        }));
-        return;
-      }
-
-      if (key === "scale") {
-        if (value.constructor === Array) {
-          value = value[0] + ',' + value[1]
-        }
-        this.setState({ scale: value });
-        return;
-      }
-
-      let newState = {};
-      if (typeof key === "string") {
-        newState[key] = value;
-      } else {
-        for (let i = 0; i < key.length; i++) {
-          newState[key[i]] = value[i];
-        }
-      }
-      this.setState(newState);
+  const updatePlotTitle = (title) => {
+    if (title !== state.plotTitle) {
+      setState((prev) => ({ ...prev, plotTitle: title }));
     }
-  }
-
-  compareChanged(checked) {
-    let newScale = checked ? "-10,10,auto" : this.props.dataset_0.variable_scale + ",auto"
-    this.setState({ scale: newScale });
-    this.props.setCompareDatasets(checked)
-  }
-
-  onTabChange(index) {
-    this.setState({
-      currentTab: index,
-    });
-  }
-
-  render() {
-    _("Dataset");
-    _("Time");
-    _("Start Time");
-    _("End Time");
-    _("Variable");
-    _("Variable Range");
-    _("Colourmap");
-    _("Show Bathymetry Contours");
-    _("Arrows");
-    _("Additional Contours");
-    _("Show Selected Area(s)");
-    _("Saved Image Size");
-
-    const plotOptions = (
-      <div>
-        {/* Image Size Selection */}
-        <ImageSize
-          key="size"
-          id="size"
-          state={this.state.size}
-          onUpdate={this.onLocalUpdate}
-          title={_("Saved Image Size")}
-        ></ImageSize>
-
-        {/* Plot Title */}
-        <CustomPlotLabels
-          key="title"
-          id="title"
-          title={_("Plot Title")}
-          updatePlotTitle={this.updatePlotTitle}
-          plotTitle={this.state.plotTitle}
+  };
+
+  const compareChanged = (checked) => {
+    const newScale = checked
+      ? "-10,10,auto"
+      : props.dataset_0.variable_scale + ",auto";
+    setState((prev) => ({ ...prev, scale: newScale }));
+    props.setCompareDatasets(checked);
+  };
+
+  const onTabChange = (index) =>
+    setState((prev) => ({ ...prev, currentTab: index }));
+
+  // Translation keys for side-effects (if needed)
+  useEffect(() => {
+    [
+      "Dataset",
+      "Time",
+      "Start Time",
+      "End Time",
+      "Variable",
+      "Variable Range",
+      "Colourmap",
+      "Show Bathymetry Contours",
+      "Arrows",
+      "Additional Contours",
+      "Show Selected Area(s)",
+      "Saved Image Size",
+    ].forEach((key) => t(key));
+  }, [t]);
```

Comment:
I think this was only necessary inside the `render` function. It would be worth checking if it can be omitted once you switch to the `_()` shortcut.